pennylane/devices/qubit/apply_operation.py

```diff
@@ -263,6 +263,52 @@ def apply_cnot(op: qml.CNOT, state, is_state_batched: bool = False, debugger=Non
     return math.stack([state[sl_0], state_x], axis=control_axes)
 
 
+@apply_operation.register
+def apply_multicontrolledx(
+    op: qml.MultiControlledX, state, is_state_batched: bool = False, debugger=None
+):
+    r"""Apply MultiControlledX to state by composing transpositions, rolling of control axes
+    and the CNOT logic above."""
+    if len(op.wires) < 8:
+        return apply_operation_einsum(op, state, is_state_batched)
+
+    ctrl_wires = [w + is_state_batched for w in op.hyperparameters["control_wires"]]
+    # apply x on all "wrong" controls
+    roll_axes = [
+        w for val, w in zip(op.hyperparameters["control_values"], ctrl_wires) if val == "0"
+    ]
+    state = math.roll(state, 1, roll_axes)
+
+    orig_shape = math.shape(state)
+    # Move the axes into the order [(batch), other, target, controls]
+    transpose_axes = (
+        np.array(
+            [w for w in range(len(orig_shape)) if w not in op.wires]
+            + [op.total_wires[-1]]
+            + op.total_wires[:-1].tolist()
+        )
+        + is_state_batched
+    )
+    # transpose_axes =  + is_state_batched for w in transpose_axes]
+    state = math.transpose(state, transpose_axes)
+
+    # Reshape the state into 3-dimensional array with axes [batch+other, target, controls]
+    state = state.reshape((-1, 2, 2 ** (len(op.wires) - 1)))
```

Comment:
```suggestion
    state = math.reshape(state, (-1, 2, 2 ** (len(op.wires) - 1)))
```

Tensrflow tensors do not have a `reshape` method